### ISSUE TITLE
Temperature limitations?

### ISSUE THREAD

**Issue by Marinela-os** [beliveau-lab/PaintSHOP_pipeline#15]
First of all, thank you so much for creating this pipline and your efforts to improve and simplify probe design strategies.

I tested the pipeline and generated newBalance DNA probes for chromosomes of interest using the default settings in config.yml file, changing only the max and min. oligo-length. I work on a plant species and I used soft masked draft genome and accompaniying annootation file.
However, after taking into account protocol conditions and changing Tm:
blockparse_min_length: 42
blockparse_max_length: 48
blockparse_min_tm: 75
blockparse_max_tm: 90

search results in no probes fund and empty intermediate files.  As maximum temperature option for XGBoost is 60 - I started wondering if there is a limit for Tm range that can be incorporated?
If yes, what would be an option for desiging probes used in protocols that call for hybridization at high temperatures?

Best,
Marinela

**Comment by brianbeliveau** [comment]
Hi Marinela,

Those temperatures seem very high—are you already accounting for the presence of formamide in the experiment? Can you please list your desired experimental hybridization conditions (salt, formamide, temperature) so we can see if we do have a suitable XGBoost model pre-trained that would suit your needs?

**Comment by Marinela-os** [comment]
Hi Brian,
Thank you for your fast reply and you are absolutely right - Tm is too high! Since we are working with a non-model organism and have only a genome assembly draft available, I am performing the computational part of this project while a cytologist from the group that is working on protocol development. It seems after revision that we will go for 42°C as hybridization temperature, and I used lower Tms accordingly (42-49°C). Now all works like a charm, and even after extensive filtering of NewBalance dataset, we are left with more than 100,000 DNA probes (for 17 Mb chromosome) - that should give us a lot of freedom in setting up the desired probe density. 
Thank you once again for the tool and your help!

p.s. As i do not want to spam with issues, I will ask for additional clarification here...
After obtaining a set of probes, I understood from the 2021 publication that one should be able to work with subsets of probes to visualize distribution and append the primers in PainSHOP app. However, it is not clear where can one upload a generated set of probes and/or assembly for the species of interest. Did I get that wrong? - Is web application only available for pre-defined datasets (all seem to be model-organisms)? 

Best,
Marinela

**Comment by conorcamplisson** [comment]
Hi Marinela, 

I'm glad to hear you're able to get probes from the pipeline with your params now!

It is true that the instance of the PaintSHOP web app hosted at [paintshop.io](https://paintshop.io/) only works with probe collections that we manually upload to the back-end (an AWS S3 bucket), which the app then pulls down on-demand. 

However, the shiny app is also open source (repo [here](https://github.com/beliveau-lab/PaintSHOP)) and you could run the app locally (eg from within RStudio, after cloning the repo and installing R dependencies). The next step would be to modify the app's source code to include the location of your new probe files. These files could be stored locally or in the cloud. You can inspect [this](https://github.com/beliveau-lab/PaintSHOP/commit/de732fdd1e76703ce27ead84277c17269fb9ad0e) recent commit which shows the changes to the shiny app's source code required to make it aware of a new probe collection. These are AWS S3 URLs but you could also use paths to local files on the machine running the shiny app.

If you do want to go this route and you have any questions, please feel free to reach out to me at `concamp@uw.edu`

Best,

Conor

**Comment by brianbeliveau** [comment]
Hi Marinela, we may be able to host probe collection for your assembly on paintshop. Please email us with the details of what you are running (input files, organism) and we can take a closer look to see what would be involved.

**Comment by Marinela-os** [comment]
This is wonderful! Thank you both for your help! 
I will first run the app locally and keep you informed (and probably bother with some additional Qs!)
I have to discuss with other project members about having a probe collection (I guess publicly) on paintshop. I can defitely see how that would be super useful in the long run, but as I said we are just developing the protocol atm, everything is unpublished and quality has to be assed better. I will e-mail you :)
Thank you immensely for everything!
Best,
Marinela
